src/scr_flush_async_bbapi.c

```diff
@@ -112,14 +112,17 @@ static int scr_flush_async_file_dequeue(scr_hash* hash1, scr_hash* hash2)
 static int bb_check(int rc)
 {
   if (rc) {
-    char* errstring;
-    getLastErrorDetails(BBERRORJSON, &errstring);
-    printf("Error rc:       %d\n", rc);
-    printf("Error details:  %s\n", errstring, errstring);
-    free(errstring);
-
-    //printf("Aborting due to failures\n");
-    //exit(-1);
+    int ret;
+    size_t l_NumBytesAvailable;
+    ret = BB_GetLastErrorDetails(BBERRORJSON, &l_NumBytesAvailable, 0, NULL);
+    if(rc == 0)
+    {
+        char* errstring = malloc(l_NumBytesAvailable+1);
+        BB_GetLastErrorDetails(BBERRORJSON, NULL, l_NumBytesAvailable, &errstring);
+        printf("Error rc:       %d\n", rc);
+        printf("Error details:  %s\n", errstring, errstring);
+        free(errstring);
```

Comment:
and let's replace free(errstring) with scr_free(&errstring)

note we pass the address of the pointer in here, because the function will set it back to NULL after freeing the memory